frontend/javascripts/oxalis/model/sagas/min_cut_saga.js

```diff
@@ -0,0 +1,722 @@
+// @flow
+import _ from "lodash";
+
+import type { Action } from "oxalis/model/actions/actions";
+import { type Saga, call, put, select } from "oxalis/model/sagas/effect-generators";
+import { V3 } from "libs/mjs";
+import type { Vector3 } from "oxalis/constants";
+import { addUserBoundingBoxAction } from "oxalis/model/actions/annotation_actions";
+import {
+  enforceActiveVolumeTracing,
+  getActiveSegmentationTracingLayer,
+} from "oxalis/model/accessors/volumetracing_accessor";
+import { finishAnnotationStrokeAction } from "oxalis/model/actions/volumetracing_actions";
+import { getResolutionInfo } from "oxalis/model/accessors/dataset_accessor";
+import { takeEveryUnlessBusy } from "oxalis/model/sagas/saga_helpers";
+import BoundingBox from "oxalis/model/bucket_data_handling/bounding_box";
+import * as Utils from "libs/utils";
+import api from "oxalis/api/internal_api";
+import window from "libs/window";
+
+// By default, a new bounding box is created around
+// the seed nodes with a padding. Within the bounding box
+// the min-cut is computed.
+const DEFAULT_PADDING = [50, 50, 50];
+
+// Voxels which are close to the seeds must not be relabeled.
+// Otherwise, trivial min-cuts are performed which cut right
+// around one seed.
+// For seeds that are very close to each other, their distance
+// overrides this threshold.
+const MIN_DIST_TO_SEED = 30;
+
+const TimeoutError = new Error("Timeout");
+const PartitionFailedError = new Error(
+  "Segmentation could not be partioned. Zero edges removed in last iteration. Probably due to nodes being too close to each other? Aborting...",
+);
+
+// If the min-cut does not succeed after 10 seconds
+// in the selected mag, the next mag is tried.
+const MIN_CUT_TIMEOUT = 10 * 1000;
+// During the refinement phase, the timeout is very forgiving.
+// Even if the refinement is slow, we typically don't want to
+// abort it, since the initial min-cut has already been performed.
+const MIN_CUT_TIMEOUT_REFINEMENT = 10 * 60 * 1000;
+
+// To choose the initial mag, a voxel threshold is defined
+// as a heuristic. This avoids that an unrealistic mag
+// is tried in the first place.
```

Comment:
> Or does this result in worse cuts?

Yes, I think, when taken to the extreme (think of mag 1024), the "fat erasing" will definitely cause harm as it can easily erase too much. Still, there are probably ways to improve the heuristic, but I think for the first version, this should be fine.